source/ceylon/time/iso8601/parseTimeZone.ceylon

```diff
@@ -1,200 +1,66 @@
-import ceylon.time.base { ms = milliseconds }
 import ceylon.time.timezone {
-    timeZone,
-    TimeZone
+    TimeZone,
+    OffsetTimeZone
+}
+import ceylon.time.base {
+    milliseconds
 }
-
-
 
 "Timezone offset parser based on ISO-8601, currently it accepts the following 
  time zone offset patterns:
  
- - &plusmn;`[hh]:[mm]`, 
+ - &plusmn;`[hh]:[mm]`,
  - &plusmn;`[hh][mm]`, and 
  - &plusmn;`[hh]`.
  
  In addition, the special code `Z` is recognized as a shorthand for `+00:00`"
-shared TimeZone|ParserError parseTimeZone( String offset ) {
-    variable TzState state = TzInitial();
-    for( character in offset ) {
-        state = state.next( Chunk( character ) );
+shared TimeZone? parseTimeZone( String offset ) {
+    value signal = offset.startsWith("+") || offset.equalsIgnoringCase("Z")
```

Comment:
I would short-cirquit the special case  of  "Z" and always return ufc timezone.

Then the rest of of the stuff will be easier and more declarative to check.